edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudProxy.cs

```diff
@@ -102,17 +102,27 @@ public async Task<bool> OpenAsync()
             catch (Exception ex)
             {
                 Events.ErrorOpening(this.clientId, ex);
+                await this.HandleException(ex);
                 throw;
             }
         }
 
         public async Task<IMessage> GetTwinAsync()
         {
             this.timer.Reset();
-            Twin twin = await this.client.GetTwinAsync();
-            Events.GetTwin(this);
-            IMessageConverter<Twin> converter = this.messageConverterProvider.Get<Twin>();
-            return converter.ToMessage(twin);
+            try
+            {
+                Twin twin = await this.client.GetTwinAsync();
+                Events.GetTwin(this);
+                IMessageConverter<Twin> converter = this.messageConverterProvider.Get<Twin>();
+                return converter.ToMessage(twin);
+            }
+            catch (Exception ex)
+            {
+                Events.ErrorGettingTwin(this, ex);
+                await this.HandleException(ex);
```

Comment:
It's ok to await in a catch? Just wondering.

Reply:
Yeah, it was enabled in C# 6.0.